src/transformers/cache_utils.py

```diff
@@ -62,16 +65,16 @@ def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
         # TODO: deprecate this function in favor of `cache_position`
         raise NotImplementedError("Make sure to implement `get_seq_length` in a subclass.")
 
-    def get_max_length(self) -> Optional[int]:
```

Comment:
We will need a deprecation cycle for `get_max_length` before we can delete it :P

(in the raised warning, make sure you mention that users should use `get_max_cache_shape` instead)